lib/utils.js

```diff
@@ -68,45 +41,68 @@ const getDataStats = (data, checksum, dedupHash) => ({
   size: Buffer.byteLength(data),
 });
 
-const compress = (file, minCompressSize, compression, cb) => {
-  fs.stat(file, (err, stats) => {
-    if (err || stats.size <= minCompressSize) {
-      cb(err, false);
-      return;
-    }
-    const filec = file + 'z';
-    const compressor = compressors[compression];
-    if (!compressor) {
-      throw new Error(`Unknown compression type ${compression} specified`);
+const mkdirRecursive = async dir => {
+  try {
+    await fs.access(dir);
+  } catch (err) {
+    if (err.code === 'ENOENT') {
+      await mkdirRecursive(path.dirname(dir));
+      await fs.mkdir(dir);
+    } else {
+      throw new Error(`Cannot access directory '${dir}': ${err.message}`);
     }
-    compressor
-      .compressFile(file, filec)
-      .then(() =>
-        fs.rename(filec, file, err => {
-          if (err) cb(err, false);
-          else cb(null, true);
-        })
-      )
-      .catch(err => cb(err, false));
-  });
+  }
 };
 
-const uncompress = (file, opts, cb) => {
-  fs.access(file, err => {
-    if (err) {
-      cb(err);
-      return;
-    }
-    const compressor = compressors[opts.compression];
-    if (!compressor) {
-      throw new Error(`Unknown compression type ${opts.compression} specified`);
-    }
+const rmdirRecursive = async dir => {
+  try {
+    await fs.access(dir);
+  } catch (err) {
+    if (err.code === 'ENOENT') return;
+    throw new Error(`Cannot access directory '${dir}': ${err.message}`);
+  }
+
+  const files = await fs.readdir(dir);
+
+  for (let file of files) {
+    file = path.join(dir, file);
+    const stats = await fs.stat(file);
+
+    if (stats.isDirectory()) await rmdirRecursive(file);
+    else await fs.unlink(file);
+  }
+
+  await fs.rmdir(dir);
+};
```

Comment:
I don't like the idea of synchronous removal here, perhaps we could just use `rimraf`? Also, it handles specific codes like `EBUSY`, `EPERM` etc.

Reply:
I'd prefer to not install additional packages.

Reply:
@SemenchenkoVitaliy can you extract this function to `@metarhia/common`?

Reply:
@belochub Sure, but in `async/await` or callback style? I ask this because there are only `callback` style functions in [`@metarhia/common`](https://github.com/metarhia/common).

Reply:
Done in https://github.com/metarhia/common/pull/290.